force-app/main/default/classes/CarTriggerHelper.cls

```diff
@@ -0,0 +1,37 @@
+public with sharing class CarTriggerHelper {
+    
+    public static void updateCar(Map<Id,Product2> oldCars){
+        
+        List<Product2> updatedCars=[SELECT Id,Car_Status__c,Condition__c,Name,reservedInOpportunity__c FROM Product2 WHERE Id IN:Trigger.new];
```

Comment:
You already have all this fields in Trigger.new, so you don't have to query it again

Reply:
List<Product2> updatedCars = (List<Product2>)Trigger.new;